gen/list.go

```diff
@@ -177,3 +181,40 @@ func (l *listGenerator) Equals(g Generator, spec *compile.ListSpec) (string, err
 
 	return name, wrapGenerateError(spec.ThriftName(), err)
 }
+
+func (l *listGenerator) zapMarshaler(
+	g Generator,
+	root *compile.ListSpec,
+	fieldValue string,
+) (string, error) {
+	name := zapperName(g, root)
+	if err := g.EnsureDeclared(
+		`
+			<$zapcore := import "go.uber.org/zap/zapcore">
+
+			type <.Name> <typeReference .Type>
+			<$l := newVar "l">
+			<$v := newVar "v">
+			<$enc := newVar "enc">
+			// MarshalLogArray implements zapcore.ArrayMarshaler, enabling
+			// fast logging of <.Name>.
+			func (<$l> <.Name>) MarshalLogArray(<$enc> <$zapcore>.ArrayEncoder) error {
+				for _, <$v> := range <$l> {
+					<zapEncodeBegin .Type.ValueSpec ->
+						<$enc>.Append<zapEncoder .Type.ValueSpec>(<zapMarshaler .Type.ValueSpec $v>)
```

Comment:
@abhinav This is fine, but it has the minor downside of bailing out on the first error. This potentially leaves other errors lurking later in the array. How complex would it be to use `multierr` here and collect all the errors in one pass?

Completely fine if we leave this for a follow-up (or never do it).

Reply:
I think it'll be a bit complex because of the different corner cases with some
encoders returning errors, some not.

I'll create a follow-up task to explore that in the future but for v1 of this,
we can skip it IMO.